test/gateway/safe_core/commit_state_root.js

```diff
@@ -0,0 +1,189 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const SafeCore = artifacts.require("./SafeCore.sol");
+const MockMembersManager = artifacts.require('MockMembersManager.sol');
+const web3 = require('../../test_lib/web3.js');
+const BN = require('bn.js');
+const Utils = require('../../../test/test_lib/utils');
+const EventDecoder = require('../../test_lib/event_decoder.js');
+
+const zeroBytes =
+  "0x0000000000000000000000000000000000000000000000000000000000000000";
+
+contract('SafeCore.commitStateRoot()', function (accounts) {
+
+  let remoteChainId,
+    blockHeight,
+    stateRoot,
+    membersManager,
+    safeCore,
+    owner,
+    worker;
+
+  beforeEach(async function () {
+
+    owner = accounts[2];
+    worker = accounts[3];
+    remoteChainId = new BN(1410);
+    blockHeight = new BN(5);
+    stateRoot = web3.utils.sha3("dummy_state_root");
+    membersManager = await MockMembersManager.new(owner, worker);
+
+    safeCore = await SafeCore.new(
+      remoteChainId,
+      blockHeight,
+      stateRoot,
+      membersManager.address,
+    );
+
+    stateRoot = web3.utils.sha3("dummy_state_root_1");
+    blockHeight = blockHeight.addn(1);
+
+  });
+
+  it('should fail when state root is zero', async () => {
+
+    stateRoot = zeroBytes;
+
+    await Utils.expectRevert(
+      safeCore.commitStateRoot(
+        blockHeight,
+        stateRoot,
+        { from: worker },
+      ),
+      'State root must not be zero.',
+    );
+
+  });
+
+  it('should fail when block height is less than the latest committed ' +
+    'state root\'s block height', async () => {
+
+    blockHeight = new BN(4);
+
+    await Utils.expectRevert(
+      safeCore.commitStateRoot(
+        blockHeight,
+        stateRoot,
+        { from: worker },
+      ),
+      'Given block height is lower or equal to highest committed state root block height.',
+    );
+
+  });
+
+  it('should fail when block height is equal to the latest committed ' +
+    'state root\'s block height', async () => {
+
+    blockHeight = new BN(5);
```

Comment:
this should not be necessary. it could even harm the test if the block height is changed in `beforeEach`